File: client/src/ts/model/message.ts
import { User } from "./user";

export interface Message {
  user: User;
  isYour: boolean;
  text: string;
  date: Date;
  type: MessageType;
}

export enum MessageType {
  JOIN = "JOIN",
  CHAT = "CHAT",
  DISCONNECT = "DISCONNECT",
}

File: backend/chat/src/main/java/pl/gebka/chat/controller/model/Message.java
package pl.gebka.chat.controller.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

import java.util.Date;

@Builder
@AllArgsConstructor
@Getter
public class Message {
    private User user;
    private String text;
    private Date date;
    private MessageType type;
}
